elf_version.h

```diff
@@ -0,0 +1,132 @@
+ * so that it can be used in simple if directives by the macro preprocessor.
+ *
+ * The Mission Version number shall be set to zero in all officially released
+ * packages, and is entirely reserved for the use of the mission. The Mission
+ * Version is provided as a simple macro defined in the cfe_platform_cfg.h
+ * header file.
+ *
+ * <H2> Version Number Flexibility </H2>
+ *
+ * The major number may increment when there is no breaking change to the API,
+ * if the changes are significant enough to warrant the same level of attention
+ * as a breaking API change.
+ *
+ * The minor number may increment when there have been no augmentations to the
+ * API, if changes are as significant as additions to the public API.
+ *
+ * The revision numbers may increment in implementations where no actual
+ * implementation-specific code has changed, if there are other changes within
+ * the release with similar significance.
+ *
+ * <H2> How and Where Defined </H2>
+ *
+ * The Major, Minor, and Revision components of the version are provided as
+ * simple macros defined in the cfe_version.h header file as part of the API
+ * definition; these macros must expand to simple integer values, so that they
+ * can be used in simple if directives by the macro preprocessor.
+ *
+ * The Mission Version is provided as a simple macro defined in the
+ * cfe_platform_cfg.h header file. As delivered in official releases, these
+ * macros must expand to simple integer values, so that they can be used in
+ * simple macro preprocessor conditions, but delivered code should not prevent a
+ * mission from, for example, deciding that the Mission Version is actually a
+ * text string.
+ *
+ * <H2> Identification of development builds </H2>
+ *
+ * In order to distinguish between development versions, we provide a BUILD_NAME
+ * and a BUILD_NUMBER.
+ *
+ * The BUILDNAME is a unifying monicker that indicates compatibility across the
+ * core Flight System's propduct suite. The build name is a star or
+ * constellation name and changes once per official release. Subsequent build
+ * names starts with the next letter in the 24-character alphabet. In between
+ * official releases, the BUILDNAME will be held constant.
+ *
+ * The BUILDNUMBER reflects the number of commits since the last official
+ * release for each particular component. This integer monotonically increases
+ * for a given BUILDNAME. When a new BUILDNAME is applied, the BUILDNUMBER
+ * resets to zero.
+ */
+
+
+/*
+**  File: elf_version.h
+**
+**  Purpose:
+**  Provide version identifiers for the ELF to cFE Table Converter.
+**
+*/
+
+#ifndef _elf_version_
+#define __elf_version_
+
+/*
+** Development build identifiers
+*/
+#define ELF_BUILD_NAME   "BOOTES"
+#define ELF_BUILD_NUMBER 1
+
+
+/*
+** Macro Definitions
+*/
+#define ELF_MAJOR_VERSION          3
+#define ELF_MINOR_VERSION          1
+#define ELF_REVISION               5
+#define ELF_MISSION_REV            0
```

Comment:
Likewise all these should be `ELF2CFETBL_xxx` not `ELF_xxx`.  The latter makes me think it is referring to the ELF standard itself, which is version 1.2.

Reply:
I actually had it as elf2cfetbl_etc and changed it because there were some ELF prefixes in other variables and I didn't realize it was related to the ELF standard itself (ie. ELF_Structures.h)

Reply:
what do you recommend I use instead of underscore for the include guard name?

Reply:
nevermind, I'm going with the suggestion in the link posted